File: C++/SelectionSort.cpp
#include<iostream>
using namespace std;

void swap(int &a, int &b) 
{  int temp;
   temp = a;
   a = b;
   b = temp;
}

void display(int *arr, int n) {
   for(int i = 0; i<n; i++)
   {	cout << arr[i] << " ";
   }
   cout << endl;
}

void SelectionSort(int *arr, int n) {
   int i, j, i_min;
   for(i = 0; i<n-1; i++) 
   {  i_min = i; 
      for(j = i+1; j<n; j++)
      {   if(arr[j] < arr[i_min])
             i_min = j;
         swap(arr[i], arr[i_min]);
      }
   }
}
int main() {
   int n;
   cout << "Enter the number of elements: ";
   cin >> n;
   int array[n];         
   cout << "Enter elements:" << endl;
   for(int i = 0; i<n; i++) {
      cin >> array[i];
   }
  
   SelectionSort(arr, n);
   cout << "Array after Sorting: ";
   display(arr, n);
}

File: Go/SelectionSort.go
package main

import (
    "fmt"
)

func selectionSort(arr []int) {

	for i, x := range arr[:len(arr)-1] {
        k := i
        for j, y := range arr[i+1 : len(arr)] {
            if x > y {
                k = i + 1 + j
                x = y
            }
        }
        arr[k], arr[i] = arr[i], x
    }
}

func main() {

    array := []int{5, 8, 4, 1, 7,  2, 3, 6}
    fmt.Println("Unsorted array: ", array)
    selectionSort(array)
    fmt.Println("Sorted array: ", array)
}